File: bot/src/main/java/by/yayauheny/tutochkatgbot/dto/backend/SubwayLineResponseDto.java
package by.yayauheny.tutochkatgbot.dto.backend;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;
import java.util.UUID;

/**
 * Subway line response DTO
 */
public record SubwayLineResponseDto(
    @JsonProperty("id") UUID id,
    @JsonProperty("cityId") UUID cityId,
    @JsonProperty("nameRu") String nameRu,
    @JsonProperty("nameEn") String nameEn,
    @JsonProperty("shortCode") String shortCode,
    @JsonProperty("hexColor") String hexColor,
    @JsonProperty("isDeleted") Boolean isDeleted,
    @JsonProperty("createdAt") Instant createdAt
) {}

File: backend/src/main/kotlin/yayauheny/by/model/dto/SubwayLineResponseDto.kt
package yayauheny.by.model.dto

import java.time.Instant
import java.util.UUID
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class SubwayLineResponseDto(
    @Contextual val id: UUID,
    @Contextual val cityId: UUID,
    val nameRu: String,
    val nameEn: String,
    val shortCode: String? = null,
    val hexColor: String,
    val isDeleted: Boolean,
    @Contextual val createdAt: Instant
)
